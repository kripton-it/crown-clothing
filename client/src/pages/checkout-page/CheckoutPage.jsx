import React from "react";

import CheckoutItemContainer from "../../components/checkout-item/CheckoutItemContainer";
import StripeCheckoutButton from "../../components/stripe-button/StripeButton";

import {
  CheckoutPageContainer,
  CheckoutHeader,
  HeaderItem,
  TotalContainer,
  WarningText
} from "./CheckoutPageStyles";

const CheckoutPage = ({ cartItems, total }) => {
  return (
    <CheckoutPageContainer>
      <CheckoutHeader>
        <HeaderItem>
          <span>Product</span>
        </HeaderItem>
        <HeaderItem>
          <span>Description</span>
        </HeaderItem>
        <HeaderItem>
          <span>Quantity</span>
        </HeaderItem>
        <HeaderItem>
          <span>Price</span>
        </HeaderItem>
        <HeaderItem>
          <span>Remove</span>
        </HeaderItem>
      </CheckoutHeader>
      {cartItems.map(cartItem => (
        <CheckoutItemContainer key={cartItem.id} cartItem={cartItem} />
      ))}
      <TotalContainer>
        <span>Total: ${total}</span>
      </TotalContainer>
      <WarningText>
        *Please use the following test credit card for payments*
        <br />
        4242 42424 4242 4242 - Exp: 01/20 - CVV: 123
      </WarningText>
      <StripeCheckoutButton price={total} />
    </CheckoutPageContainer>
  );
};

export default CheckoutPage;
