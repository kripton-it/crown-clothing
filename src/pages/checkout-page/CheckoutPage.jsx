import React from "react";
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";

import CheckoutItem from "../../components/checkout-item/CheckoutItem";
import StripeCheckoutButton from "../../components/stripe-button/StripeButton";

import {
  selectCartItems,
  selectCartTotalPrice
} from "../../redux/cart/cart-selectors";

import {
  CheckoutPageContainer,
  CheckoutHeader,
  HeaderItem,
  TotalContainer,
  WarningText
} from "./CheckoutPageStyles";

const CheckoutPage = ({ cartItems, total }) => {
  return (
    <CheckoutPageContainer>
      <CheckoutHeader>
        <HeaderItem>
          <span>Product</span>
        </HeaderItem>
        <HeaderItem>
          <span>Description</span>
        </HeaderItem>
        <HeaderItem>
          <span>Quantity</span>
        </HeaderItem>
        <HeaderItem>
          <span>Price</span>
        </HeaderItem>
        <HeaderItem>
          <span>Remove</span>
        </HeaderItem>
      </CheckoutHeader>
      {cartItems.map(cartItem => (
        <CheckoutItem key={cartItem.id} cartItem={cartItem} />
      ))}
      <TotalContainer>
        <span>Total: ${total}</span>
      </TotalContainer>
      <WarningText>
        *Please use the following test credit card for payments*
        <br />
        4242 42424 4242 4242 - Exp: 01/20 - CVV: 123
      </WarningText>
      <StripeCheckoutButton price={total} />
    </CheckoutPageContainer>
  );
};

const mapStateToProps = createStructuredSelector({
  cartItems: selectCartItems,
  total: selectCartTotalPrice
});

export default connect(mapStateToProps)(CheckoutPage);
