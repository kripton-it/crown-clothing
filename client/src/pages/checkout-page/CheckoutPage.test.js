import React from "react";
import { shallow } from "enzyme";

import CheckoutPage from "./CheckoutPage";

it("expect to render CheckoutPage page", () => {
  const mockProps = {
    cartItems: [
      {
        id: 1,
        imageUrl: "abc.com",
        name: "Hello",
        price: 100,
        quantity: 2
      }
    ],
    total: 100
  }
  expect(shallow(<CheckoutPage {...mockProps} />)).toMatchSnapshot();
});
