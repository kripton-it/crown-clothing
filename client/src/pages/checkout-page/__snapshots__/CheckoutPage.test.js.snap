// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render CheckoutPage page 1`] = `ShallowWrapper {}`;
